src/app/_components/Search.tsx

```diff
@@ -0,0 +1,74 @@
+import { useRouter } from 'next/navigation';
+import { useRef, useState } from 'react';
+
+import Button from '@/components/Button';
+import Icon from '@/components/Icon';
+import { useMediaQuery } from '@/hooks/useMediaQuery';
+import { cn } from '@/utils/cn';
+
+export default function Search() {
+  const isMobile = useMediaQuery('mobile');
+  const [isFocused, setIsFocused] = useState(false);
+  const [query, setQuery] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const router = useRouter();
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const handleClick = () => {
+    if (!query.trim()) return;
+    setIsLoading(true);
+    router.push(`/search?query=${encodeURIComponent(query)}`);
+  };
+
+  const handleWrapperClick = (e: React.MouseEvent<HTMLDivElement>) => {
```

Comment:
❗ 이전에 `react`에서 import 하는 것들은 따로 구조 분해로 쓰기로 결정한 것으로 기억합니다!
`import { MouseEvent } from 'react';`
```suggestion
  const handleWrapperClick = (e: MouseEvent<HTMLDivElement>) => {
```